host_applications/linux/apps/raspicam/RaspiVid.c

```diff
@@ -2106,6 +2117,19 @@ static MMAL_STATUS_T create_encoder_component(RASPIVID_STATE *state)
       }
    }
 
+   if (state->encoding == MMAL_ENCODING_H264 &&
+       state->slices > 1)
+   {
+      int slice_row_mb = (state->height/16)/state->slices + (state->height/16)%state->slices?1:0; //round up
```

Comment:
I assume you're trying to ensure that you round the height up such that it divides by slice_row_mb to give exactly state->slices slices of data. I don't think this calculation works correctly though.
Pass a height of 130 and 2 slices in, and I think you'll get 3 slices out each time (slice_mb_rows=4 for 64 lines per slice). The same applies to heights of 129-143.
I know these weird rounding conditions are often nasty, which is why I'm not immediately making a suggestion of a fix.

Reply:
Ooops, I forgot about non mod16 video sizes.

The usual method would probably be to round up (video height / 16) to get macroblock rows per frame, then round up (macroblock rows per frame / slice count) to get macroblock rows per slice. This should ensure we never have more slices than specified.